js/graphql.md

```diff
@@ -0,0 +1,1374 @@
+
+Here is a complete list of searchable operations per GraphQL type supported as of today:
+
+| GraphQL Type        | Searchable Operation           |
+|-------------:|:-------------|
+| String      | `ne`, `eq`, `match`, `matchPhrase`, `matchPhrasePrefix`, `multiMatch`, `exists`, `wildcard`, `regexp` |
+| Int     | `ne`, `gt`, `lt`, `gte`, `lte`, `eq`, `range`      |
+| Float | `ne`, `gt`, `lt`, `gte`, `lte`, `eq`, `range`      |
+| Boolean | `eq`, `ne`      |
+
+#### Generates
+
+Todo: What does searchable generate.
+
+## Examples
+
+### Simple Todo
+
+```
+type Todo @model {
+  id: ID!
+  name: String!
+  description: String
+}
+```
+
+### Blog
+
+```
+type Blog @model {
+  id: ID!
+  name: String!
+  posts: [Post] @connection(name: "BlogPosts")
+}
+type Post @model {
+  id: ID!
+  title: String!
+  blog: Blog @connection(name: "BlogPosts")
+  comments: [Comment] @connection(name: "PostComments")
+}
+type Comment @model {
+  id: ID!
+  content: String
+  post: Post @connection(name: "PostComments")
+}
+```
+
+#### Blog Queries
+
+```
+# Create a blog. Remember the returned id.
+# Provide the returned id as the "blogId" variable.
+mutation CreateBlog {
+  createBlog(input: {
+    name: "My New Blog!"
+  }) {
+    id
+    name
+  }
+}
+
+# Create a post and associate it with the blog via the "postBlogId" input field.
+# Provide the returned id as the "postId" variable.
+mutation CreatePost($blogId:ID!) {
+  createPost(input:{title:"My Post!", postBlogId: $blogId}) {
+    id
+    title
+    blog {
+      id
+      name
+    }
+  }
+}
+
+# Create a comment and associate it with the post via the "commentPostId" input field.
+mutation CreateComment($postId:ID!) {
+  createComment(input:{content:"A comment!", commentPostId:$postId}) {
+    id
+    content
+    post {
+      id
+      title
+      blog {
+        id
+        name
+      }
+    }
+  }
+}
+
+# Get a blog, its posts, and its posts comments.
+query GetBlog($blogId:ID!) {
+  getBlog(id:$blogId) {
+    id
+    name
+    posts(filter: {
+      title: {
+        eq: "My Post!"
+      }
+    }) {
+      items {
+        id
+        title
+        comments {
+          items {
+            id
+            content
+          }
+        }
+      }
+    }
+  }
+}
+
+# List all blogs, their posts, and their posts comments.
+query ListBlogs {
+  listBlogs { # Try adding: listBlog(filter: { name: { eq: "My New Blog!" } })
+    items {
+      id
+      name
+      posts { # or try adding: posts(filter: { title: { eq: "My Post!" } })
+        items {
+          id
+          title
+          comments { # and so on ...
+            items {
+              id
+              content
+            }
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+### Task App
```

Comment:
Can we add a little prose here "The following is a tutorial of using the transformer to build a backend for a task management application that supports...."